### PR TITLE
Friends app

### PR DESCRIPTION
**Pull request by Zihfred**
_Complete the form below and remove this line._

### Friends app

[Demo](https://zihfred.github.io/Friends-App/) |
[Code base](https://github.com/Zihfred/Friends-App)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Zihfred/Friends-App/script.js [hunk 1]
```diff
@@ -0,0 +1,270 @@
+class FriendApp {
+    constructor() {
+        this.sortBox = document.querySelector('.sortBox');
+        this.peoples = document.querySelector('.peoples');
+        this.resetBtn = document.querySelector('.sortBox__resetBtn');
+        this.sortInput = document.querySelector('.sortBox__sortedInput');
+        this.sortIconMale = document.querySelector('.fa-male');
+        this.sortIconFemale = document.querySelector('.fa-female');
+        this.data = [];
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/script.js]: `data` very generic name. `friends` maybe?

File: submissions/Zihfred/Friends-App/script.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            })
+    }
+    generatePeoples(data) {
+        data.forEach(onePeople => {
+            this.addCardToPeoples(this.generateOnePeopleCard(onePeople));
+        })
+    }
+
+    makeFirstLetterUpper(str) {
+        return str[0].toUpperCase() + str.slice(1);
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/script.js]: `text-transform: uppercase`

File: submissions/Zihfred/Friends-App/script.js [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+        let fullName = document.createElement('div');
+        fullName.classList.add('onePeople__fullName');
+        fullName.classList.add('onePeople__description');
+        fullName.textContent = `${this.makeFirstLetterUpper(obj.name.first)} ${this.makeFirstLetterUpper(obj.name.last)}`;
+
+        let location = document.createElement('div');
+        location.classList.add('onePeople__description');
+        location.classList.add('onePeople__location');
+        location.textContent = `${this.makeFirstLetterUpper(obj.location.city)}`;
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/script.js]: `this.makeFirstLetterUpper` will return string, so you don't need `${}` for such cases

File: submissions/Zihfred/Friends-App/script.js [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        let age = document.createElement('div');
+        age.classList.add('onePeople__description');
+        age.classList.add('onePeople__age');
+        age.textContent = `Возраст: ${(obj.dob.age)}`;
+
+        onePeople.appendChild(avatar);
+        onePeople.appendChild(fullName);
+        onePeople.appendChild(location);
+        onePeople.appendChild(age);
+        onePeople.appendChild(number);
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/script.js]: You can use `ParentNode.append` for such cases and deal all of this with one call

File: submissions/Zihfred/Friends-App/script.js [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+    addCardToPeoples(card) {
+        this.peoples.appendChild(card);
+    }
+
+    clearPeoplesForm() {
+        this.peoples.innerHTML = '';
+    }
+
+    sort(e) {
+        if (e.target.classList.contains('fas')) {
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/script.js]: What is `fas`? 🤔

File: submissions/Zihfred/Friends-App/script.js [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+                return -1;
+            }
+            return 0;
+        });
+        this.clearPeoplesForm();
+        this.generatePeoples(res);
+
+
+    }
+    filterByMale(){
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/script.js]: IMO, you can make one function `filterByGender`

File: submissions/Zihfred/Friends-App/script.js [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+        this.sortIconFemale.classList.add('disabled');
+        this.sortIconMale.classList.add('nonActive');
+        if(this.sortInput.value){
+            this.data = this.filterByInputAll('male');
+        }else{
+            this.data = this.startData;
+        }
+
+        this.data = this.data.filter(function (people) {
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/script.js]: You can use arrow function here. Will make code looks cleaner

File: submissions/Zihfred/Friends-App/script.js [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+
+        this.clearPeoplesForm();
+        this.generatePeoples(this.data);
+    }
+    filterByInputAll(gender){
+        let result;
+
+            result = this.data.filter(function (people) {
+
+                return people.name.first.indexOf(this.sortInput.value) != -1 || people.name.last.indexOf(this.sortInput.value) != -1;
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/script.js]: `String#includes`

File: submissions/Zihfred/Friends-App/script.js [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+
+    addEventsListeners() {
+        this.sortInput.addEventListener('input', this.renderSortedByInput);
+        this.sortBox.addEventListener('click', this.sort);
+        this.resetBtn.addEventListener('click', this.resetPeople);
+
+    };
+}
+
+new FriendApp();
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/script.js]: Pls add line in the end of a file

File: submissions/Zihfred/Friends-App/style.css [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+.nonActive{
+    pointer-events: none;
+}
+
+
+
+
+
+/*sort box */
+.sortBox {
```
Comment (zonzujiro) [inline submissions/Zihfred/Friends-App/style.css]: Pls, give class names with `kebab-case` or `snake_case`. `camelCase` usually only for JS.

**Comment by zonzujiro** [comment]
Usability issues:

image

As you see, items will be filtered only when I click on an icon. But cursor says that I can click near an icon and if I will do so - nothing happened. You should fix this.

Also, button `Reset` is redundant. I think you don't need it. Can you remove it, how you think?

File: submissions/Zihfred/Friends-App/script.js (hunk 5 above)
Comment (Zihfred) [reply submissions/Zihfred/Friends-App/script.js]: <i class="fas fa-male" data-type="male"></i>

Comment (Zihfred) [reply submissions/Zihfred/Friends-App/script.js]: i class="fas fa-male" data-type="male"

**Comment by zonzujiro** [comment]
@Zihfred pls, resolve all outdated discussions when you done

**Comment by zonzujiro** [comment]
Hey! You did a great job worth special recognition.

Here is your achievement badge:
[Kottans-Student-Course-progress-5%3D-Course%20Top--5%20performer-green.svg)](https://github.com/kottans/kottans/blob/master/endorsements.md)

Congratulations on the skills level-up!

**Comment by zonzujiro** [comment]
Ok, looks like I have made mistake and gave badge little too [soon](https://github.com/kottans/frontend-2019-homeworks/pull/71) :) 

I will not withdraw it, but you must remember, that all finished task is required for getting access to offline part of the course. So, please finish unfinished :)

**Pull request merged**